job-runner/chain/chain.go

```diff
@@ -145,19 +122,24 @@ func (c *Chain) IsDoneRunning() (done bool, complete bool) {
 		case proto.STATE_RUNNING:
 			// If any jobs are still running, the chain isn't done or complete.
 			return false, false
-		case proto.STATE_PENDING, proto.STATE_STOPPED:
-			// If any job can be run, the chain is not done or complete.
-			// Treat stopped jobs as pending jobs because they may be retried,
-			// as when resuming a suspended job chain.
+		case proto.STATE_STOPPED:
+			// Stopped jobs are not runnable in this context (i.e. chain context).
+			// Do not return early here; we need to keep checking other jobs.
+		case proto.STATE_PENDING:
+			// If any job is runnable, the chain isn't done or complete.
 			if c.isRunnable(job.Id) {
 				return false, false
 			}
-		default:
-			// Any job that matches none of the above cases is failed
+			// This job is pending but not runnable which means a previous job
+			// failed.
+		case proto.STATE_FAIL:
+			// If sequence can retry, then chain isn't done or complete,
 			if c.canRetrySequence(job.Id) {
-				// This failed job is part of a sequence that can be retried.
 				return false, false
 			}
+			// Failed but no seq retry means the chain has failed
+		default:
+			panic("IsDoneRunning: invalid job state: " + proto.StateName[job.State])
 		}
 
 		// We can only arrive here if a job is not complete (pending or failed).
```

Comment:
Update this comment, we also arrive here if a job is stopped.

Reply:
Done.